### PR TITLE
fix: Disable knative liveness and readiness probes

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- We are seeing issues were knative readiness and liveness probs are require to use port 8080. If port 8080 is being utilized we see that user will have issues with karpenter start up. https://github.com/aws/karpenter/issues/4145

**How was this change tested?**
- Manually tested 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/operator/operator.go [hunk 1]
```diff
@@ -156,8 +155,13 @@ func (o *Operator) WithControllers(ctx context.Context, controllers ...corecontr
 func (o *Operator) WithWebhooks(ctx context.Context, webhooks ...knativeinjection.ControllerConstructor) *Operator {
 	if !injection.GetOptions(ctx).DisableWebhook {
 		o.webhooks = append(o.webhooks, webhooks...)
-		lo.Must0(o.Manager.AddReadyzCheck("webhooks", knativeChecker("readiness")))
-		lo.Must0(o.Manager.AddHealthzCheck("webhooks", knativeChecker("health")))
+		go func(ctx context.Context) {
```
Comment (njtran) [inline pkg/operator/operator.go:158]: Do we need some backoff here? Does this just aggressively spam the "waiting" log line?

**Review by njtran — COMMENTED** [review]
one comment about backoff

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -175,21 +174,18 @@ func (o *Operator) Start(ctx context.Context) {
...
 
-func knativeChecker(path string) healthz.Checker {
+func webhookChecker(ctx context.Context) healthz.Checker {
 	return func(req *http.Request) (err error) {
-		res, err := http.Get(fmt.Sprintf("http://localhost:%d/%s", knativeinjection.HealthCheckDefaultPort, path))
+		_, err = http.Get(fmt.Sprintf("http://localhost:%d", injection.GetOptions(ctx).WebhookPort))
 		if err != nil {
 			return err
 		}
-		if res.StatusCode != http.StatusOK {
```
Comment (njtran) [inline pkg/operator/operator.go:190]: nit: Can you add a TODO to add this back in when we're able to configure the health check ports?

File: pkg/operator/operator.go [hunk 3]
```diff
@@ -175,21 +174,19 @@ func (o *Operator) Start(ctx context.Context) {
...
 	}
 	wg.Wait()
 }
 
-func knativeChecker(path string) healthz.Checker {
+func webhookChecker(ctx context.Context) healthz.Checker {
+	// TODO: Add knative health check port for webhooks when health port can be configured
 	return func(req *http.Request) (err error) {
-		res, err := http.Get(fmt.Sprintf("http://localhost:%d/%s", knativeinjection.HealthCheckDefaultPort, path))
+		_, err = http.Get(fmt.Sprintf("http://localhost:%d", injection.GetOptions(ctx).WebhookPort))
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:186]: Should we check for 500s and report liveness failures if we are getting 500s?

File: pkg/operator/operator.go [hunk 4]
```diff
@@ -175,20 +175,22 @@ func (o *Operator) Start(ctx context.Context) {
...
 }
 
-func knativeChecker(path string) healthz.Checker {
+func webhookChecker(ctx context.Context) healthz.Checker {
+	// TODO: Add knative health check port for webhooks when health port can be configured
+	// Issue: https://github.com/knative/pkg/issues/2765
 	return func(req *http.Request) (err error) {
-		res, err := http.Get(fmt.Sprintf("http://localhost:%d/%s", knativeinjection.HealthCheckDefaultPort, path))
+		res, err := http.Get(fmt.Sprintf("http://localhost:%d", injection.GetOptions(ctx).WebhookPort))
 		if err != nil {
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:189]:
```suggestion
		if err != nil || res.StatusCode >= 500 {
```

File: pkg/operator/operator.go [hunk 5]
```diff
@@ -175,20 +175,22 @@ func (o *Operator) Start(ctx context.Context) {
...
 			return err
 		}
-		if res.StatusCode != http.StatusOK {
-			return fmt.Errorf("%s probe failed, %s", path, lo.Must(io.ReadAll(res.Body)))
+		if res.StatusCode == http.StatusInternalServerError {
```
Comment (jonathan-innis) [inline pkg/operator/operator.go:192]:
```suggestion
                 Or there is a server-side error, consider liveness to have failed
		if res.StatusCode >= 500 {
```

**Review by jonathan-innis — APPROVED** [review]
Nice 🎉

**Pull request merged**